### ISSUE TITLE
Predictions: More than one or no providers are configured... error when using yarn link to link predictions from forked amplify-js repo

### ISSUE THREAD

**Issue by ryanhollander** [aws-amplify/amplify-js#9337]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Vue

### Amplify APIs

Authentication, Predictions

### Amplify Categories

auth, predictions

### Environment information

<details>

```
# Put output below this line
  System:
    OS: Windows 10 10.0.19042
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 16.67 GB / 31.93 GB
  Binaries:
    Node: 16.13.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.1.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (96.0.1054.43)
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    <%= pkg.name %>:  <%= pkg.version %>
    @apollo/client: ^3.3.21 => 3.5.6
    @apollo/client/cache:  undefined ()
    @apollo/client/core:  undefined ()
    @apollo/client/errors:  undefined ()
    @apollo/client/link/batch:  undefined ()
    @apollo/client/link/batch-http:  undefined ()
    @apollo/client/link/context:  undefined ()
    @apollo/client/link/core:  undefined ()
    @apollo/client/link/error:  undefined ()
    @apollo/client/link/http:  undefined ()
    @apollo/client/link/persisted-queries:  undefined ()
    @apollo/client/link/retry:  undefined ()
    @apollo/client/link/schema:  undefined ()
    @apollo/client/link/utils:  undefined ()
    @apollo/client/link/ws:  undefined ()
    @apollo/client/react:  undefined ()
    @apollo/client/react/components:  undefined ()
    @apollo/client/react/context:  undefined ()
    @apollo/client/react/hoc:  undefined ()
    @apollo/client/react/hooks:  undefined ()
    @apollo/client/react/parser:  undefined ()
    @apollo/client/react/ssr:  undefined ()
    @apollo/client/testing:  undefined ()
    @apollo/client/testing/core:  undefined ()
    @apollo/client/utilities:  undefined ()
    @apollo/client/utilities/globals:  undefined ()
    @aws-amplify/analytics: ^5.0.4 => 5.1.8
    @aws-amplify/api: ^4.0.4 => 4.0.28
    @aws-amplify/api-graphql: ^2.0.4 => 2.2.17
    @aws-amplify/core: ^4.1.2 => 4.3.10
    @aws-amplify/datastore: ^3.1.2 => 3.7.2
    @aws-amplify/predictions: ^4.0.28 => 4.0.28
    @aws-amplify/pubsub: ^4.0.4 => 4.2.4
    @aws-amplify/ui-components: ^1.6.0 => 1.9.6
    @aws-amplify/ui-vue: ^1.0.13 => 1.1.20
    @babel/eslint-parser: ^7.13.14 => 7.16.3
    @babel/plugin-transform-runtime: ^7.14.5 => 7.16.4
    @babel/polyfill: ^7.12.1 => 7.12.1
    @babel/runtime: ^7.14.6 => 7.16.3
    @quasar/app: ^3.2.5 => 3.2.5
    @quasar/extras: ^1.12.2 => 1.12.2
    @speechly/browser-client: ^1.0.19 => 1.0.19
    @speechly/speech-recognition-polyfill: ^1.0.0 => 1.1.0
    @vue/apollo-composable: ^4.0.0-alpha.13 => 4.0.0-alpha.16
    @vue/devtools: ^5.3.4 => 5.3.4
    HelloWorld:  0.0.1
    amplify-ui-components-loader:  undefined ()
    apollo-cache-inmemory: ^1.6.6 => 1.6.6 (1.3.12)
    apollo-client: ^2.6.10 => 2.6.10 (2.4.6)
    apollo-link-http: ^1.5.17 => 1.5.17 (1.5.8)
    aws-amplify: ^4.1.2 => 4.3.10
    aws-appsync: ^4.0.5 => 4.1.4
    aws-appsync-auth-link: ^3.0.5 => 3.0.7 (2.0.7)
    axios: ^0.21.1 => 0.21.4
    core-js: ^3.6.5 => 3.19.3 (2.6.12)
    crypto-browserify: ^3.12.0 => 3.12.0
    crypto-js: ^4.1.1 => 4.1.1
    envinfo: ^7.8.1 => 7.8.1
    eslint: ^7.14.0 => 7.32.0
    eslint-config-prettier: ^8.1.0 => 8.3.0
    eslint-plugin-quasar: ^1.0.0 => 1.0.0
    eslint-plugin-vue: ^7.0.0 => 7.20.0
    eslint-webpack-plugin: ^2.4.0 => 2.6.0
    fs: ^0.0.1-security => 0.0.1-security
    graphql: ^15.5.1 => 15.8.0 (14.5.0, 14.0.0)
    graphql-tag: ^2.12.5 => 2.12.6
    hermes-inspector-msggen:  1.0.0
    mitt: ^3.0.0 => 3.0.0
    node-polyfill-webpack-plugin: ^1.1.4 => 1.1.4
    quasar: ^2.3.4 => 2.3.4
    react-native: ^0.64.2 => 0.64.3
    vue-apollo: ^3.0.7 => 3.1.0
    vuex: ^4.0.2 => 4.0.2
    workbox-webpack-plugin: ^6.0.0 => 6.4.2
    xml-js: ^1.6.11 => 1.6.11
  npmGlobalPackages:
    @aws-amplify/cli: 7.5.5
    @quasar/cli: 1.2.2
    @vue/cli: 4.5.15
    yarn: 1.22.17

```

</details>

### Describe the bug

In my app, when I Predictions.convert() get 'More than one or no providers are configured, Either specify a provider name or configure exactly one provider' if I use the steps in the contribution guide to link the @aws-amplify/predictions from my amplify-js source. FWIW I've tried this with the clean main branch and with my passed all the build and test steps modified from the main branch. If I unlink the repo it works fine. Just trying to clear the last hurdle before a pull request. :-) 

### Expected behavior

It should work / play the text.

### Reproduction steps

fork/clone amplify-js
follow the contribution guide steps to link @aws-amplify/predictions
run a sample app

(sorry I can't share my app)

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by ryanhollander** [comment]
I should add... I'm trying to test a change to the Polly integration.

**Comment by jamesaucode** [comment]
👋  Hi there, if you yarn link the entire `aws-amplify` does it work?
Assuming you already ran `yarn setup-dev` on your forked repo folder.

in your app, `yarn link aws-amplify`

import it like this `import { Predictions } from 'aws-amplify'`

**Comment by ryanhollander** [comment]
hi @jamesaucode Thanks for responding. I did try linking the whole package and it elicited the same error. 

I've never run 'yarn setup-dev' b/c I was not aware of that step. FWIW: I followed the contributor instructions here: https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md

So I tried that, re-did the build, linking, yada yada. Same issue.

Undid the linking. redid the linking to aws-amplify as you describe and now I cannot login to my app at all, I get an auth error and it acts like it does not have aws-exports.js.

Any advice?

**Comment by chrisbonifacio** [comment]
Hey @ryanhollander 👋 are you still experiencing issues with linking that package and/or in need of assistance?

**Comment by ryanhollander** [comment]
@chrisbonifacio yes. I just tried it again. I update the predictions package in the amplify-js source. I run the build/test/linking steps. I yarn link @aws-amplify/predictions in my working with the current release version env / code. I get this when I run in the browser: 

Uncaught (in promise) Error: More than one or no providers are configured, Either specify a provider name or configure exactly one provider
    at PredictionsClass.getPluggableToExecute (Predictions.ts?a39f:204:1)
    at PredictionsClass.convert (Predictions.ts?a39f:156:1)
    at TTSEngineIntegrationProxy_CLASS.eval [as _fetchAudioStreamPromise] (polly-tts-proxy.js?e0ea:12:1)
    at TTSEngineIntegrationProxy_CLASS.playAudioFromPlainText (TTSEngineIntegrationProxy_CLASS.js?3269:53:1)
    at Object.func (var-handler.js?f308:14:1)
    at Object.processTemplateEntry (say-response.js?21dc:43:1)
    at Object.playAudioResponse (say-response.js?21dc:17:1)
    at async EventHandler.eval [as _handler] (speech-response-say-handler.js?6680:11:1)
    at async EventHandler.handle (EventHandler_CLASS.js?e641:25:1)

Any guidance you may have is appreciated. I'd really like to get these features in the main line so we can use them and get my feet wet contributing to Amplify.

**Comment by ryanhollander** [comment]
I'm closing this. It looks like the features I wanted to add are available through the GraphQL / API integration for Predictions which must have been released sometime back and I missed it. I'll refactor my app to use that. I've tried 17 different things to get this linking scheme to work to no avail. I'm bailing now, sorry! I'll come back to it when I have some time to dig in / contribute. Thanks for your help!
